### PR TITLE
Fix for FGOALS-f3-L sftlf

### PR DESCRIPTION
**Pull request by mwjury**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValCore/blob/master/CONTRIBUTING.md).

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [ ] Add unit tests
-   [ ] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [ ] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [ ] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

### CODE REVIEW

File: esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Fixes for FGOALS-f3-L model."""
+from ..fix import Fix
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py:2]:
```suggestion
from ..fix import Fix
import dask.array as da
```

File: esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        ----------
+        cube: iris.cube.Cube
+            Cube to fix
+
+        Returns
+        -------
+        iris.cube.Cube
+            Fixed cube. It can be a difference instance.
+        """
+        if cube.units == "%" and cube.data.max() <= 1.:
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py:23]:
```suggestion
        if cube.units == "%" and da.max(cube.core_data()).compute() <= 1.:
```

File: esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            cube.data = cube.data * 100
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py:24]:
```suggestion
            cube.core_data = cube.core_data() * 100.
```

**Comment by mwjury** [comment]
@valeriupredoi Thanks for the code review.
The main problem I have is that the fix does not get applied and hence the error mentioned in #667 gets thrown. I am using the sftlf data for the mask_landsea, but it does not get corrected.

**Comment by valeriupredoi** [comment]
right - the problem is that currently the mask files that go to `mask_*` preprocessors don't get fixed on the fly - the only code patch that does that is in #439 but that is awfully off-track from `master` - for now the only suggestion I have is to use `sftlf` as a regular variable (hence your fix will be applied in that case) and use its (fixed) file in the diagnostic rather than in the preprocessor - can you do that or is it off limits of your workflow? :beer:

**Comment by mwjury** [comment]
@valeriupredoi doh, thanks for clarifying!
Nah, that would mean a very long extra way I'd have to go in the diagnostics. Any estimate yet as to when #439 will be merged? Maybe it's best to wait a bit.

**Comment by valeriupredoi** [comment]
#439 will never be merged, it'll die alone :grin: we are waiting for iris 3.0 to have a fully functional cell measures handling implemented so fx data can be handled seamlessly by them - these fixes are still useful though; you could fix the data via asking for sftlf as a regular variable, save the fixed files and then use them with a custom data input dir for the actual run; twisty two-stage pipeline but will get you to where you need to be :beer:

**Comment by mwjury** [comment]
haha, I see. Thanks for the hint, let's see if I can may this work (or the modellers are changing the data first).

**Comment by valeriupredoi** [comment]
modellers changing data is a very long wheelbase truck as @zklaus would confirm :grin:

**Comment by jvegasbsc** [comment]
The codacy issue is added by yapf, so I left it alone

**Comment by jvegasbsc** [comment]
@valeriupredoi , I think we can merge this now, no?

**Review by valeriupredoi — APPROVED** [review]

**Comment by valeriupredoi** [comment]
@jvegasbsc be my guest :+1:

**Pull request merged**